### ISSUE TITLE
Enhancement: Smoothing castbar animation

### ISSUE THREAD

**Issue by artemicon** [shagu/pfUI#115]
Is it possible to edit the castbar animation?  Right now it looks blocky and jumpy when casting.  I would like to smooth it out so that it behaves like the unit frames.

**Comment by shagu** [comment]
What do you mean by "blocky and jumpy"? For me, the animation runs smooth.

**Comment by artemicon** [comment]
I was testing using a low cast-time spell like polymorph.  When I cast instead of fluidly animating the cast bar it ticks over every .1 second.  I checked and my blizzard castbar and ecasting bar also do the same.  If this is just normal behavior for vanilla, I'm sorry.  I'n wow now a days it animates fluidly the space between the .1 so that it doesn't appear as a tick but a fluid transition.

**Comment by Cruzix** [comment]
@artemicon Its smooth in the current version however I also noticed what you're saying in earlier versions. Just update and you should be good.

**Comment by artemicon** [comment]
I updated the the latest build and I still see the same behavior.  I made a video showing it.  I tested with polymorph and the hearthstone.  Longer spells look more fluid because it fits more ticks into a smaller space.

https://youtu.be/qaYQFkoP1Go

**Comment by shagu** [comment]
I tried to reproduce this with many different settings (double buffering, triple buffering, vsync, ...)
But sadly I was not able to reproduce this. For me, it is always really smooth. 
Could you please add some information about your system (operating system, graphic card, ..)?

**Comment by artemicon** [comment]
Thanks for replying to this Shagu, I appreciate you even looking at this.  When you mentioned changing settings I remembered I ran the commands listed in this web page to try and 'enhance' my wow experience.  I don't know if it will have any affect on that but I thought I would let you know.

https://forum.nostalrius.org/viewtopic.php?f=32&t=1100

I'm on Windows 10 with a GTX 680.

Thanks a ton!

**Comment by shagu** [comment]
I just tried these settings on my client, but still my animations are smooth. 
As the castbar uses GetTime() to check for its progress, it could be related to this part:
>timingModeOverride 1 (uses GetTickCount) 2 (uses RDTSC) 3 (uses QueryPerformanceCounter) 4 (uses timeGetTime). Experiment with which timing method gives you the smoothest gameplay. Set the value, then restart the game completely. Apparently RDTSC is the most precise timing method.

Could you please check that?

**Comment by shagu** [comment]
Well guys, I need your help here.
Please open the file modules/castbar.lua and insert after line 156 the following:
`          message(GetTime())
`

It should look like this afterwards:
```
        -- cast
        if ( pfUI.castbar.player.casting ) then
          **message(GetTime())**
          local status = GetTime()
          local cur = round(GetTime() - pfUI.castbar.player.startTime,1)
```

Reload your UI (/rl) and as soon as you start casting something, all GetTime() ticks will be shown in the left chatframe. In these values, I'm very interested in. Please take a screenshot them.

Thanks in advance.

**Comment by artemicon** [comment]
At work at the moment and have some errands to do as soon as I go home so I won't get to this until late this evening.  Will update with findings, though.

**Comment by artemicon** [comment]
Okay I tried the timingModeOverride different settings logging in and out between settings.  No luck.  I made a video with the gettime output I got while casting a hearthstone.  I don't have a chat addon to copy otherwise I would have to make it easier.

https://youtu.be/iVvnAKBi_Pc

I tried a TBC and WOTLK client and both of their cast bars are functioning normally.

**Comment by artemicon** [comment]
I also booted into my linux drive (Ubuntu MATE 16.04) and ran the same executable I'm using in windows through wine and it works perfectly...

**Comment by artemicon** [comment]
I think I found my problem.  I booted back into windows this morning and after some time noticed my clock was out of sync (I'm assuming it was to do with booting into my linux OS).  I decided to check wow before correcting it and low and behold it works perfectly.  My automatic time settings were unchecked so I rechecked them and it corrected them and it still works.

So I believe that it was something to do with the time sync of my windows 10 OS install.  I'm glad I got it fixed and sorry for leading you on a wild goose hunt.
